docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx

```diff
@@ -30,87 +33,108 @@ To complete the steps in this guide, you'll need:
 
 ## Step 1: Generate a Dagster+ agent token
 
+<Note>
+  <strong>If using Serverless deployment</strong>, skip this step.
+</Note>
+
 <GenerateAgentToken />
 
 ---
 
 ## Step 2: Create and configure an agent
 
 <Note>
-  <strong>
-    If using{" "}
-    <a href="/dagster-plus/deployment/serverless">Serverless deployment</a>
-  </strong>
-  , this step isn't required.
+  <strong>If using Serverless deployment</strong>, skip this step.
 </Note>
 
 <BDCreateConfigureAgent />
 
 ---
 
-## Step 3: Add GitHub workflow files to your repository
+## Step 3: Set up the GitHub Action files
 
 <Note>
-  If you used the GitHub app to add the repository as a code location, this step
-  isn't required.{" "}
+  <strong>If you used the Dagster+ GitHub app to import your project</strong>,
+  this step was already done for you.{" "}
   <a href="#step-4-configure-the-github-repository">Skip to the next step</a>.
 </Note>
 
-In this step, you'll add the required GitHub workflow files to your GitHub repository. The files can be found in our [Hybrid quickstart repository](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart) or [Serverless quickstart repository](https://github.com/dagster-io/dagster-cloud-serverless-quickstart), depending on the agent you are using.
+In this step, you'll add the required GitHub Action files to your GitHub repository. The files can be found in the [Hybrid quickstart repository](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart) or [Serverless quickstart repository](https://github.com/dagster-io/dagster-cloud-serverless-quickstart), depending on the deployment type you're using.
 
-Copy the following files to your repository (the linked files are shown in the Hybrid repo, there are equivalents in the Serverless repo).
+<TabGroup>
+<TabItem name="Hybrid">
 
-- [`dagster_cloud.yaml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/dagster_cloud.yaml)
-- [`.github/workflows/dagster-cloud-deploy.yml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/.github/workflows/dagster-cloud-deploy.yml)
+### Hybrid
```

Comment:
TLDR; at the end

I find the UI experience of adding a code location in Hybrid kinda confusing. We ask users to add the code location in a YAML editor, which contains the same properties as a `dagster_cloud.yaml`. Since there isn't a connected repository, specifying this in Dagster+ doesn't automatically create/add a `dagster_cloud.yaml` to the project.

So a user then has to manually create the `dagster_cloud.yaml` themselves with the same info they specify using the YAML editor - right? 

I haven't tested this, but if someone tries to add a code location/project using the CLI but there isn't a `dagster_cloud.yaml`, what happens?

TLDR: This feels like a weird experience and it's proving difficult to describe here and in other guides.

Reply:
hmmm I don't have a compete answer for this either. 

One thing that could help is that in this page https://docs.dagster.io/dagster-plus/managing-deployments/code-locations we say that using a `dagster_cloud.yaml` file is recommended for code locations in the first section, but then there isn't a section later in the page that describes adding/modifying/deleting code locations when you're using `dagster_cloud.yaml`

Reply:
I think we should steer away from recommending people use the UI to add a code location, for the reason that you mention, especially in a guide about CI/CD. They shouldn't need to do that at all for this to work.